src/Hydra/JsonSchema/SchemaFactory.php

```diff
@@ -133,14 +131,24 @@ public function buildSchema(string $className, string $format = 'jsonld', string
 
         $prefix = $this->getSchemaUriPrefix($schema->getVersion());
         $collectionKey = $schema->getItemsDefinitionKey();
-        $key = $schema->getRootDefinitionKey() ?? $collectionKey;
+
+        $definitionName = $this->definitionNameFactory->create($className, $format, $inputOrOutputClass, $operation, $serializerContext);
+        if (Schema::TYPE_INPUT === $type) {
+            $definitionName .= '.input';
+        }
+
+        $jsonSchema = $this->schemaFactory->buildSchema($className, 'json', $type, $operation, new Schema(version: $schema->getVersion()), $serializerContext, $forceCollection);
```

Comment:
this doesn't work as references will be with `json` instead of `jsonld`

Reply:
But then relations references are not correct anymore

Reply:
@soyuka Sorry, I don't understand what you're saying. Could you please explain in more detail? 🙇‍♂️

Reply:
IMO it is a really hard problem to fix I spent quite some time in the refactoring trying not to break this :/